### ISSUE TITLE
ExpressionVisitor CCE using package qualifier on generic function w/constraints

### ISSUE THREAD

**Issue by jvasileff** [ceylon/ceylon-spec#1347]
This used to work:

```ceylon
T id<T>(T t) given T satisfies Object => nothing;
String s = package.id("");
```
I now get:
```
the 'ExpressionVisitor' caused an exception visiting a 'InvocationExpression' node: '"java.lang.ClassCastException"'
```
In my original code (non-simplified), I get:
```
the 'ExpressionVisitor' caused an exception visiting a 'InvocationExpression' node: '"java.lang.ClassCastException: com.redhat.ceylon.model.loader.model.LazyPackage cannot be cast to com.redhat.ceylon.model.typechecker.model.TypeDeclaration"' at 'com.redhat.ceylon.model.typechecker.model.Type.getTypedMember(Type.java:1227)'
```

**Comment by gavinking** [comment]
On the branch, you mean? I can't reproduce it on master.

**Comment by jvasileff** [comment]
Ok, good, me neither. Looks like it was fixed in the last 12hrs or so.

**Comment by jvasileff** [comment]
Ha, so not a false report. Just a *very* quick fix in 92ac480e42d191c0779414336fbc1b637c3e4159
